### ISSUE TITLE
Newsstand iOS5 integration

### ISSUE THREAD

**Issue by AvenNigera** [Simbul/baker#100]
Hey Guys,

Does anyone know how "Newsstand" of iOS 5 could work?

Newsstand
A custom newsstand for all your subscriptions.
Read all about it. All in one place. iOS 5 organizes your magazine and newspaper app subscriptions in Newsstand: a folder that lets you access your favorite publications quickly and easily. There’s also a new place on the App Store just for newspaper and magazine subscriptions. And you can get to it straight from Newsstand. New purchases go directly to your Newsstand folder. Then, as new issues become available, Newsstand automatically updates them in the background — complete with the latest covers. It’s kind of like having the paper delivered to your front door. Only better.

We would like to use baker/laker but we need a "folder" which will include all magazines which anyone can download every month a new mag for free or for some cents. The question is does anyone know that we can develop or build the mag with laker/baker and use the "Newsstand" App when its published to get the functions of abonnements and the possibilities to download a new one in the background, to buy a new one, to have a list of all in one folder (not every magazine is one app) and so on?

Thanks alot

Aven

**Comment by folletto** [comment]
We are following that part, and we'll integrate in Baker as much as possible as soon as we get details about Newsstand - and iOS5 will be released (due to the legal restrictions of the prerelease of iOS5). :)

**Comment by AvenNigera** [comment]
This sounds awesome! Thanks :)

**Comment by jayohTM** [comment]
^ bump. certainly interested in this.

**Comment by Virtualsweden** [comment]
Ios 5 out now.. :-)

**Comment by folletto** [comment]
Reference links:
- http://developer.apple.com/library/ios/#documentation/StoreKit/Reference/NewsstandKit_Framework/_index.html#//apple_ref/doc/uid/TP40010838
- https://support.urbanairship.com/customer/portal/articles/140033-adding-newsstand-support-to-your-app

**Comment by folletto** [comment]
Ok, Newsstand is implemented in the current master (future 3.1), to show up the magazine in the Newsstand folder, it was quite simple, some documentation is on the wiki:
https://github.com/Simbul/baker/wiki/Newsstand-support

For the full Newsstand support (background downloads) we need first to implement the multi-book issue / shelf support, Issue #20.

So, even if now Baker apps will appear in Newsstand, we keep this Issue open until we have the full support.

**Comment by nin9creative** [comment]
Do you know when there might be a release with this integration (the full multi-book / shelf / newstand integration)?  I'm currently working on a single issue book/mag but it may have the potential to be a newstand subscription down the road.

**Comment by folletto** [comment]
First we need to close Issue #20, scheduled for 4.0 release, and that's pretty big. Only after that we will be able to see if we can do the full Newsstand integration. :)
We have no date, unfortunately, since we can work on this only in our spare time.

**Comment by mutador** [comment]
Someone has created a free subscription to the Newstand? How? Baker needs to be implemented in xcode?

http://developer.apple.com/search/index.php?q=newsstand
http://developer.apple.com/library/ios/#documentation/StoreKit/Reference/NewsstandKit_Framework/_index.html
http://developer.apple.com/library/ios/#technotes/tn2280/_index.html

**Comment by tstegart** [comment]
Sooooo.... :) How much do I have to donate to move Newsstand to the top of the list. Don't joke with me, I'm serious :)

**Comment by mutador** [comment]
jajajaja   me too , i am serios too. 

newsstand is like the rain for the forest...

**Comment by folletto** [comment]
For our knowledge and prioritization...
What do you need about Newsstand exactly? :)

About Paying... we can only accept donations because they help us to keep running, but we can't provide a specific schedule since it's not our daily work (unless we get enough donations, of course).

If you'd like to pay to have some kind of delivery date however, you might hire a developer and put him in direct contact with us so we can work together. ;)

**Comment by mutador** [comment]
http://www.viggiosoft.com/blog/blog/2011/10/17/ios-newsstand-tutorial/

this is for pdf, need a visor similar in baker for newstand

if i understand well, now you can upload the separated issues and make a visor general for all of them, by integrated whit newstand....it´s no easy we are try for a weeks, but our xcode developer is limited...how my english..

**Comment by folletto** [comment]
Ok, so you are basically asking about the Shelf, not Newsstand. ;)

The Shelf is on Issue #20. :)

Newsstand support will be after the Shelf implementation, and will be basically only the background download, that's the only missing step. :)

**Comment by mutador** [comment]
no

now . if you want to put a magazine in newstand , app store ask you for a visor integrated whit newstand suport ( not external downloads, not shelf) , better if you try to upload your book to newsstand and see what happend.... i thik @tstegart  explain better

**Comment by folletto** [comment]
Uhm then I don't know, I didn't dig yet into too much detail, but from what I saw it's just an add-on to an existing shelf system, see the urban aircraft link above.
If someone has more details it would be nice. :)

**Comment by tstegart** [comment]
So Newsstand has a few different elements. One, there is code identifying your App as a Newsstand App. This, along with new icons you make, puts it in Apple's special Newsstand Store and in the Newsstand folder on the user's iPad or iPhone. 
Two, to be accepted as a Newsstand App, you have to provide for the possibility of downloading multiple issues from within the App. There is a special software kit called Newsstand Kit that you guys at Baker and Laker would need to integrate into the code, which checks for new issues, downloads books or articles in the background and tells the App and the user when it is done.

However, and here is the annoying part, every person who uses Baker would need to have their own server on which to put their issues up so the App can download it. I'm sure that would not be tough in and of itself, but each Baker user would need to go through the Baker code and add in our own server information in the right place, which is pretty complex for newbies and people who don't use XCode every day. So its probably a good idea for you guys as the developers to keep a good handle on what parts will need to be changed and document it well, or you're just going to be facing lots of questions from dorks like me. By the way, the user needs to be able to input issue names, dates and titles as well, so there is quite a bit of information that Baker users will have to enter into the code.

Lastly, and even more annoyingly, if you want to sell your books or magazine issues for money, there are additional requirements, with two aspects. Baker will need to integrate Apple's in-app purchasing software, and the Baker user will need to apply for an Apple certificate to use it. The Baker user then needs to go into the code and put his information in the right place again. This allows iPad/iPhone users to purchase some issues but not others, etc. and re-download old issues and all sorts of fun stuff. 
There are also push notifications, which alert the user to new issues being available. These are not required, but obviously helpful from a marketing perspective. They also require a certificate from Apple btw.

So that's a Newsstand summary. It seems simple from the outset, but once you dig into it its hard. On the other hand, I strongly suggest the Baker developers start work sooner rather than later on it, because I'm guessing Apple will eventually require books and magazines to use it and will start rejecting apps that don't use it. From their perspective, it would be silly to have some magazines in Newsstand and other Magazines floating around the user's iPhone. The only way to get them all in one place will be to start requiring developers to use it, and we know how Apple loves to have things all in one place. 
Tom

**Comment by folletto** [comment]
Ok, then it's exactly as I was saying:
1. **Identify the App as Newsstand**: _DONE_, it will be released as 3.1 (see above, on this thread).
2. **Download multiple issues**: _PLANNED_. It's this very issue, and it's dependant of Issue #20, 
3. **Purchasing**: _NOT planned_, and so far nobody asked yet, but I think if it's the case, it's going to be after Issue #20 and after this issue as well.
4. **Push** : _NOT planned_, nobody asked for it so far. Again it's an add-on, and not Newsstand dependent.

:)

**Comment by tstegart** [comment]
You guys are awesome :)

**Comment by nin9creative** [comment]
@folletto I think as a framework you can only go so far.  I think that if Baker supported multiple issues to download/archive/redownload and a shelf view that launched open the downloaded issues that is about as far as you can go in a generic sense.  If someone is interested in using in app purchase and push notifications for background downloading and advanced stuff that would be on the shoulders of anyone developing that stuff for their apps to run with.

I'd just be happy if we could get #2 done :)

I forked Baker and have been trying to get a combined (siteless.org shelf) working with the current Baker release.  The problem is that Baker as it stands now isn't really suited for being invoked as a viewer for a given hpub (at least from a shelf point of view).

**Comment by mutador** [comment]
only one thing  for all and for @tstegart 

http://urbanairship.com/

they do newsstand integration whit their owner hosting but we belive there are a way in itunesconnect.apple.com  when you create a newsstand and upload the binary of the diferent issue to linkage the download directly to appstore hosting, understand?

if you upload the issue to appstore you can make a visor to download direct from appstore,(whitout external hosting) we read about this but....

**Comment by folletto** [comment]
Thanks guys. getting this kind of support, suggestions and attention is AWESOME. :)

Btw, if someone thinks that the other two things above (Purchases and Push) should be done, feel free to open an issue. Push in particular should be possible to be added independently of everything else. :)

We have also the code re-factoring to allow easier embedding, but that's a really big chunk, we are trying to balance out optimizations and features right now. :)

**Comment by tstegart** [comment]
I'm in favor of purchasing, because that's how we plan to make money with our magazine (like most magazines). Without it, you would have to release each issue as a separate App, which Apple is definitely starting to frown upon. So eventually it will be needed unless people using Baker only plan on releasing one book or giving away their stuff for free. I know there are a lot of people that will be doing that so it will come first, but we will be striving to offer subscriptions.

@mutador  - I've checked out Urban Airship, the problem is with pricing. Lets say you want to sell a magazine, one issue per month and you offer a subscription for $9.99 U.S. dollars, which is reasonable. That means you will be making $7.00 after Apple's cut. In the Free Basic Urban Airship pricing, it costs $.10 per download and $.10 per user per month for subscriptions. So if you add that up it will cost you $1.20 for a user to subscribe to your magazine for a year, and $1.20 to download all 12 issues. Subtract that from $7.00 and you end up with $4.60 for each person that subscribes to your magazine for one year. If you need to pay people for content you need to get a lot of subscribers to make enough money. 
I'm not saying doing it yourself to make the extra $2.40 will be a huge difference, but it is enough to matter when you start figuring out how much we need to pay our writers and for overhead, etc.

**Comment by mutador** [comment]
vamos, que la integracion con newsstand es una cosa fundamental del proyecto, una vez estabilizada la versión, y los problemas que daba al girar etc...., ay que agusto me he quedado hablando en español

hey, integration with newsstand is a fundamental thing of the project, once stabilized version, and the problems facing the turn to and that agusto ...., etc. I stayed talking in Spanish

**Comment by tstegart** [comment]
Dude, that's totally awesome. I love it. It looks as if it's mostly working except for the UI design elements.

**Comment by AvenNigera** [comment]
This is AWESOME dude!

**Comment by folletto** [comment]
May I just ask you to proceed on Issue #20? Because here is a bit off topic, the Shelf is there. ;)

**Comment by nin9creative** [comment]
Sure!  In my head Newsstand and this is all smooshed together...  lol  Is there any way to move my comments?

Warm Regards,
Andrew

**Comment by folletto** [comment]
Awesome, thanks! :D
It's ok, nice idea of summarizing. ;)

**Comment by mutador** [comment]
I must be very short but I do not understand, in the email you send , you say that baker has a Newsstand basic support, but the reality is you can not post a magazine make with baker in the  Newstand , that asks you to programming  expecific fot it  whit a visor, a free or pay suscription, etc...

actually can not have a magazine  in the Newstand whit baker, or yes?, if it´s possible put it on the newsstand and someone made it, could explain how?

thanks

**Comment by folletto** [comment]
As you can read from the comment above, Newsstand is a set of different features:
1. The UI in the Springboard to visualize the app as a magazine
2. The Background Download feature

Currently Baker implements (1), not (2), both of them are "Newsstand". :)

**Comment by mutador** [comment]
sorry folletto but you say there are examples of baker in the appstore in newsstand? can you show me one?

**Comment by folletto** [comment]
Not that I'm aware of. But 3.1 has been released only today, so that would have been hard. :)

**Comment by mutador** [comment]
if the only thing u have done is add to the plist the line 'Application presents content in Newsstand=YES', apple will reject your app because don't have the way to manage subcriptions. I did it before, and apple reject my app.

**Comment by tstegart** [comment]
You're correct mutador. I think what they're trying to say is that they're currently still working on the other parts.

**Comment by mutador** [comment]
thanks @tstegart , and thanks @folletto.
the really for the moment  is we can not put a baker magazine in newsstand.

i´m going to open a issue to talk about profesional editorial and magazine whit baker, problems, future...., we are in spain and we like to colaborate with other people whit a professional project if this is about fashion, photo, arts...

I think baker , with their simple book currently working with does not have the same problems and needs with a large magazine

**Comment by folletto** [comment]
Do you have any evidence of a iPad Newsstand magazine rejected because it's missing a multi-issue download? We'd like to see exactly the reasoning behind that to have a clearer view. :)

@mutador please don't be misled by the default book. We use that just as a simple test, to show how simple it is to publish with the platform, it's not by any mean expression of our understanding of the full possibilities of Baker, as you can clearly see. We are well aware of the needs of professional publishing for magazine. :)

**Comment by tstegart** [comment]
While the App Store Approval guidelines don't cover the subject, the iTunes Connect developer guide ([Newsstand](http://developer.apple.com/library/mac/#documentation/LanguagesUtilities/Conceptual/iTunesConnect_Guide/14_Newsstand/Newsstand.html)) states:

> "To submit a Newsstand app, your app must be enabled for Newsstand, offer at least one auto-renewable In-
> App Purchase subscription, and the Info.plist of the associated binary must include UINewsstandApp=true. In
> addition, UINewsstandIcon must be included in the CFBundleIcons key."

That makes it sound like you need to offer subscriptions (even if they're free) to be a Newsstand app.

**Comment by mutador** [comment]
hi tstegart

https://github.com/Simbul/baker/issues/20#issuecomment-3237216

**Comment by folletto** [comment]
@tstegart Uhm, yes. If someone could try to submit a magazine with Baker 3.1 without the auto-update we can see if that will get you rejected... or not. :)

**Comment by mutador** [comment]
yes , we have been  rejected... because baker dont have implemented  a free/payment  suscription system..... and apple force you to implement this

**Comment by folletto** [comment]
Ok, thanks. :)

**Comment by mutador** [comment]
http://www.viggiosoft.com/blog/blog/categories/newsstand/

**Comment by tstegart** [comment]
Can I suggest turning off the Newsstand implementation in the current download of Baker? By that I mean set the UINewsstandApp to "NO." I suggest this because without full Newsstand integration (subscriptions and downloads) your app will always be rejected. So a person downloading Baker to make a book and forgetting to turn it off will be rejected. It is better to have it off right now so anyone using Baker doesn't have to change it before submitting. I know its seems trivial to change but for a lot of newbies its not and will end up tripping them up.

**Comment by folletto** [comment]
Yes, it's a good point. :)

**Comment by mutador** [comment]
https://github.com/DeenNa/SimpleNewsstand

**Comment by folletto** [comment]
Uhm there's no documentation or links. Do you know what it does and if there's a source?

**Comment by mutador** [comment]
no really, i only think its good to see how it works....its for pdf´s whit a external download

**Comment by folletto** [comment]
TODO:
1. Implement Newsstand Subscriptions to be a fully Newsstand app (a "Subscribe" button in the toolbar and the underlying code to manage it).
2. Implement icon switch support with the icon of the latest publication.

**Comment by folletto** [comment]
If someone has more informations on the **MINIMAL** set of features we are missing to be accepted as a Newsstand app it would be great. Right now it seems we have to create "Free subscriptions" with the app that can barely acknowledge them.

This answer: http://stackoverflow.com/questions/11895035/do-newsstand-apps-require-push-notifications

> No, Apple's guidelines do not mandate the use of push notifications, and your app should not be rejected for not using them.

Seems implying that we might not even implement a portion of that to allow Newsstand apps to exist, even if of course it would be ideal.

**Comment by nin9creative** [comment]
So, I suppose at this point instead of continuing work on my Bakershelf branch I should contribute some code back to the shelf branch!  LOL  I would spend more time trying to retrofit the newer version of Baker 3.2.2 with what I did prior.  

As for your question above, I think minimally it has to support having a Free subscription option, have all the right icons set up, updating the cover art when new issues are available.  I haven't seen a good document that outlines the minimum requirements.

However earlier on in the Bakershelf branch I think @mpena2099 was trying to get his app approved and something came up with the background downloading or resumption of downloading if you quit the app...

Would need to do some more research.  However I do know for a fact that you don't have to implement push notifications.

**Comment by Simbul** [comment]
Thanks for that insight.
Do you reckon that "free subscription" concepts has got anything to do with In-App Purchase, or is it separate based on your experience?

**Comment by nin9creative** [comment]
> 11.12
> Apps offering subscriptions must do so using IAP, Apple will share the same 70/30 revenue split with developers for these purchases, as set forth in the Developer Program License Agreement.
> 11.13
> Apps that link to external mechanisms for purchases or subscriptions to be used in the App, such as a “buy” button that goes to a web site to purchase a digital book, will be rejected
> 11.14
> Apps can read or play approved content (specifically magazines, newspapers, books, audio, music, and video) that is subscribed to or purchased outside of the App, as long as there is no button or external link in the App to purchase the approved content. Apple will not receive any portion of the revenues for approved content that is subscribed to or purchased outside of the App
> 11.15
> Apps may only use auto renewing subscriptions for periodicals (newspapers, magazines), business Apps (enterprise, productivity, professional creative, cloud storage) and media Apps (video, audio, voice), or the App will be rejected.

—

**Comment by folletto** [comment]
Thanks. For me the most important part is:

> However I do know for a fact that you don't have to implement push notifications.

Because the guidelines doesn't help really much. Like: does a "Free Subscription" require any coding at all (11.12)? The answer seems yes in theory, but in practice seems no...

**Comment by danupoyner** [comment]
I thought you'd be interested to know that I found out the reason the app was crashing (shelf branch) when I opened one of my books is because I had -baker-index-height set to 'null'. As soon as I put in a value (eg 200) it opened fine.

Also, can we confirm the final size for the book thumbnails? As far as I'm aware, Newsstand requires 100x150 images. The shelf currently uses something a bit wider than that I think.

**Comment by Simbul** [comment]
According to the [Newsstand FAQs](http://developer.apple.com/library/ios/#technotes/tn2280/_index.html#//apple_ref/doc/uid/DTS40011215-CH1-NEWSSTAND_APP_ICONS-WHAT_ARE_THE_SIZE_GUIDELINES_FOR_NEWSSTAND_APP_ICONS_), the longest edge of the Newsstand icon should be
- at least 90px for iPhone
- at least 126px for iPad
  and all high resolution images should double the size.

The [iOS Human Interface Guidelines](http://developer.apple.com/library/ios/#documentation/userexperience/conceptual/mobilehig/IconsImages/IconsImages.html#//apple_ref/doc/uid/TP40006556-CH14-SW12), somewhat confusingly, specify that Newsstand cover icons should be at least 512px on the longest edge (double for hi res). This image will then be scaled to the size specified above.

All in all, I think we'll fix this when we properly implement cover management.

**Comment by anjimi** [comment]
Hi all, I'm just catching up with where the Newsstand stuff is up to. I've been busy working on content for our magazine app which I've been testing with Baker 3.2. I'd like to try using the shelf branch now and have just downloaded the latest files from the shelf branch. Are there any instructions on how to implement and test the shelf with multiple issues? Would the first issue of our magazine be included in the books folder of our app or do I need to implement downloading of the first issue, and then subsequent issues after that?

I apologise if this information is already to be found somewhere, but I can only find mention of part of the process, and I'm not sure how to proceed.

**Comment by folletto** [comment]
It's not complete yet, but this is a good starting point, check if it works for you: https://github.com/Simbul/baker/wiki/Newsstand-vs-Bundled-publications-support-in-Baker-4.0

**Comment by anjimi** [comment]
Thanks Davide, I had seen this page but it doesn't offer any instruction on how to implement issue downloads. I saw a comment on https://github.com/Simbul/baker/issues/522 which mentions hosting a .json file on your own website, but I can't find anything further on implementation of this, and where this is referenced in the shelf app.

**Comment by folletto** [comment]
I quickly added a page, check this: https://github.com/Simbul/baker/wiki/Newsstand-shelf-JSON

**Comment by anjimi** [comment]
Great! Thank you for that, I'll try that today.

**Comment by Simbul** [comment]
To answer your other question, note that, when Newsstand is enabled, Baker will not display any of the bundled books. In fact, you can delete the `books` folder from the Xcode project entirely.

(I have added this to the wiki page as well)

**Comment by tstegart** [comment]
I love the shelf guys, great work!

Some first impressions:

When you're in a book, the button to get back to the Baker shelf says "Baker Shelf." You might want to consider changing this to something more generic, like "Library." Only because end users are not going to know what Baker is, therefore most developers will need to change it, either to eliminate the word Baker or to add in their own name. It seems uneconomical to build into the app something everyone will want to change, and I'm sure you guys will not want to answers questions on how its done every few months, which is probably inevitable. 

You might want to consider adding in an "About" button in the shelf. The shelf currently lacks any way other than books to display information to the user. At least the example I downloaded didn't have one. Is this planned? I'm thinking about privacy policies being the big issue here as something you would like the user to be able to access before they download books. But also contact info, a support email (in case the server is down and none of the books work) and just basic info on how the app works (especially what the consequences of clicking the subscribe button are). Any information that won't change from issue to issue is a good idea to display in the shelf. Are we sure an app will get accepted without this info being available? 

I suggest the about button link to a downloaded book, and if it cannot download it, it uses a bundled book (or uses a json parameter to determine what to do). This way people can design their "about" sections to match their magazines look and feel and have a way to give more information.

Questions: Can the big baker logo in the shelf be replaced with a logo of any size? Or are there dimension restrictions?
When lots of issues are added, does the shelf paginate itself, or just scroll to show them all. Is it sideways or vertical scrolling?

Some suggestions for future versions:
A preview function. A preview button opens up a modal window of a specific size that downloads a preview. It could be the same size on landscape and portrait view to same some coding trouble. If it can download an html file, people can design their own preview pages.

Thats all I have from my initial preview. Its looks great! Thanks to everyone who put in all the hard work.

**Comment by folletto** [comment]
Hi @tstegart thank you very much for the detailed feedback! That's great! :+1: 

Let's see...
1. Yes, the title will be editable or generic. It's an excellent point. :)
2. The about button isn't part of version 4.0, you're the first asking for such a thing (and there are apps in the store without it) but let's see what other people think and keep track of it: open a new issue so we have it. That will also allow anybody that wants to contribute to develop that and send us a Push Request. :)
3. Preview is something that almost no other platform provides, and it would be overkill to add right now both for us and authors, but that said it's surely an interesting feature, and it's worth its own issue. ;)

About your questions:
1. The full background is an image, so it's **COMPLETELY CUSTOM**. :)
2. The shelf scrolls, vertical. Try it. ;)

**Comment by tstegart** [comment]
Thanks @folletto!

I'll start a new issue for the About button, thanks. The preview suggestion was definitely just for the future, I don't want to delay 4.0 :)

The background is an image, but does it repeat itself when you have a lot of issues? Do you need to make the background image the same size as however many issues you have, or is it set to fill the full screen, and if you have more issues then it repeats or just ends? 

Can I ask what is left on the to-do list before the shelf is functional to be used to submit an app? I know its not done, but I'm wondering what is left? 
T

**Comment by folletto** [comment]
Ok. :)
1. The background is fixed everything scrolls above it but the top part is always free (try it), we found it's the best solution to achieve flexibility and customizability with the lowest possible effort. :)
2. For the todo list just check the milestone 4.0 here on GitHub, click "Milestones" at the top of this page. :)

**Comment by olvr** [comment]
The shelf functionality is looking great. 

It has me thinking, it looks like currently the two options you are aiming towards are 1. Full Newsstand integration and 2. Bundled publications only.

Are you thinking of allowing use of the shelf functionality in a stand alone app without any Newsstand integration too?

It would be useful to have the shelf functionality without any Newsstand integration for publications that don't meet the Newsstand requirements.

Might it also be a simpler target to include shelf functionality first then meet all the specific Newsstand requirements later?

**Comment by folletto** [comment]
We evaluated that options and while we agree it would be great... it's overkill trying to push it out right now, since it's the least required one and will require a lot more code to be written.

Shelf functionality... it's already there: the problem isn't "shelf" but "How do you manage the updates?". That's the big problem. :)

**Comment by Simbul** [comment]
Newsstand support with free subscription (and no push notifications) has been completed in c951ecae84e225d685ed3b893eba5a989c1e3203, thus I'm closing this issue :)

**Comment by tstegart** [comment]
Awwww yeeaaaahhh!
